components/dropdown/test/dropdown-content.test.js

```diff
@@ -0,0 +1,259 @@
+import '../dropdown.js';
+import '../dropdown-content.js';
+import { aTimeout, expect, fixture, html, nextFrame, oneEvent, triggerFocusFor } from '@open-wc/testing';
+
+const normalFixture = html`
+	<div>
+		<div id="optionallyFocusable">
+			<d2l-dropdown>
+				<button class="d2l-dropdown-opener"></button>
+				<d2l-dropdown-content>
+					<p id="non_focusable_inside">a</p>
+					<a id="focusable_inside" href="http://www.desire2learn.com">b</a>
+				</d2l-dropdown-content>
+			</d2l-dropdown>
+			<p id="non_focusable_outside">c</p>
+			<button id="focusable_outside">out here</button>
+		</div>
+	</div>
+`;
+
+describe('d2l-dropdown', () => {
+
+	let dropdown, content;
+
+	beforeEach(async() => {
+		dropdown = await fixture(normalFixture);
+		content = dropdown.querySelector('d2l-dropdown-content');
+	});
+
+	describe('events', () => {
+
+		it('fires open event when open attribute is added', async() => {
+			content.setAttribute('opened', true);
+			await oneEvent(content, 'd2l-dropdown-open');
+			expect(content.opened).to.be.true;
+		});
+
+		it('fires close event when open attribute is removed', async() => {
+			content.setAttribute('opened', true);
+			await oneEvent(content, 'd2l-dropdown-open');
+
+			const dropdownClose = oneEvent(content, 'd2l-dropdown-close');
+			content.removeAttribute('opened');
+			await dropdownClose;
+
+			expect(content.opened).to.be.false;
+		});
+
+	});
+
+	describe('toggleOpen', () => {
+
+		it('opens dropdown when closed', async() => {
+			content.toggleOpen();
+			await oneEvent(content, 'd2l-dropdown-open');
+			expect(content.opened).to.be.true;
+		});
+
+		it('closes dropdown when open', async() => {
+			content.toggleOpen();
+			await oneEvent(content, 'd2l-dropdown-open');
+
+			const dropdownClose = oneEvent(content, 'd2l-dropdown-close');
+			content.toggleOpen();
+			await dropdownClose;

```

Comment:
For cases like this where you need the Promise but can't `await` it until you do something else, I've seen the open-wc examples do this:

    setTimeout(() => content.toggleOpen());
    await oneEvent(content, 'd2l-dropdown-close');

That has the effect of calling `toggleOpen` _after_ the `await` line.